brave/src/test/java/brave/features/propagation/SecondaryConditionalSamplingTest.java

```diff
@@ -28,13 +29,113 @@
+ *     return (fieldName, value) -> {
+ *       // look at the extra field to see if it means we should boost the signal
+ *       if (fieldName.equals("edgar") && edgarWants(value)) builder.sampledLocal();
+ *       return value;
+ *     };
+ *   }
+ * }
+ * }</pre>
+ *
+ * <h2>Reporting data eventually to multiple systems</h2>
+ * In normal Zipkin, sampling is implicit from a backend POV. If you received data, then it must
+ * have been sampled. In code, when {@link TraceContext#sampled()} is true, data reports to the
+ * {@link Tracing.Builder#spanReporter(Reporter) normal span reporter}, and most backends assume the
+ * entire trace will be present.
+ *
+ * <p>Now, we set {@link Tracing.Builder#alwaysReportSpans()} to allow Edgar's conditionally
+ * sampled data reach Mantis. If we do nothing else, we have a concern which is that Edgar data ends
+ * up in Salp's storage even though it is incomplete from a trace POV. This could cause questions
+ * and will in any case add load to Salp's storage.
+ *
+ * <p>To make this concrete, any of the below scenarios will now result in reported spans:
+ * <ul>
+ *   <li>Salp sampled, but Edgar not</li>
+ *   <li>Salp unsampled, but Edgar sampled</li>
+ *   <li>Both sampled</li>
+ * </ul>
+ *
+ * The only trouble area is when Salp says no, but Edgar says yes. If we had more than one secondary
+ * tracing system, differentiation could be a little trick because multiple systems could trigger a
+ * {@link TraceContext#sampledLocal() sampled local decision}. Luckily, here there is no ambiguity:
+ * if {@link TraceContext#sampledLocal()} and not {@link TraceContext#sampled()}, it was Edgar!
+ *
+ * <p>Ex.
+ * <pre>{@code
+ * class AddSampledTag extends FinishedSpanHandler {
+ *   // Propagate a data routing hint to Mantis to ensure Edgar doesn't end up flooding Salp when
+ *   // Salp says no.
+ *   @Override public boolean handle(TraceContext context, MutableSpan span) {
+ *     boolean sampledToSalp = Boolean.TRUE.equals(context.sampled()); // primary remote sampling
+ *     boolean sampledToEdgar = context.sampledLocal(); // secondary overlay
+ *     span.tag("sampled", sampledToSalp && sampledToEdgar ? "salp,edgar"
+ *       : sampledToSalp ? "salp" : "edgar");
+ *     return true;
+ *   }
+ * }
+ * }</pre>
+ *
+ * <h3>Summary</h3>
+ * It is relatively easy to report more data based on a secondary decision made based on an extra
+ * field. All you need to do is implement and configure {@link ExtraFieldPropagation.Customizer} to
+ * {@link TraceContext#sampledLocal() sample local} based on the field value, and set the tracing
+ * instance to {@link Tracing.Builder#alwaysReportSpans() always report spans}. If the spans already
+ * contain data the backend needs to route to appropriate data stores, you are done. If it needs a
+ * hint, you can add a {@link FinishedSpanHandler} to add a tag which ensures the backend consistently
+ * and statelessly honors the sampling intent.
+ */
```

Comment:
haven't changed the example code, yet, but this is the design. I still need to run it through @narayaruna to ensure I am following things properly